ros/angel_system_nodes/angel_system_nodes/feedback_generator.py

```diff
@@ -0,0 +1,164 @@
+
+from angel_msgs.msg import (
+    ActivityDetection,
+    ObjectDetection3dSet,
+    TaskUpdate,
+    AruiUpdate,
+    AruiObject3d
+)
+
+
+class FeedbackGenerator(Node):
+    """
+    ROS node responsible for sending activity, detection, and task information to the ARUI when the information has updated. 
+
+    Takes in information from the `angel_msgs/ActivityDetection`, `angel_msgs/ObjectDetection3dSet`, and `angel_msgs/TaskUpdate` messages. 
+
+    Publishes `angel_msgs/AruiUpdate` representing the current activity, detections, and task.
+    """
+
+    def __init__(self):
+        super().__init__(self.__class__.__name__)
+
+        self._activity_detector_topic = self.declare_parameter("activity_detector_topic", "ActivityDetections").get_parameter_value().string_value
+        self._object_detection_topic = self.declare_parameter("object_detection_topic", "ObjectDetections3d").get_parameter_value().string_value
+        # TODO: task needs to be updated to emit TaskNode
+        self._task_monitor_topic = self.declare_parameter("task_monitor_topic", "TaskUpdates").get_parameter_value().string_value
+        self._arui_update_topic = self.declare_parameter("arui_update_topic", "AruiUpdates").get_parameter_value().string_value
+
+        # logger
+        self.log = self.get_logger()
+        self.log.info(f"Activity detector topic: {self._activity_detector_topic}")
+        self.log.info(f"Object detection topic: {self._object_detection_topic}")
+        self.log.info(f"Task monitor topic: {self._task_monitor_topic}")
+        self.log.info(f"AruiUpdate topic: {self._arui_update_topic}")
+
+        # subscribers
+        self.activity_subscriber = self.create_subscription(
+            ActivityDetection,
+            self._activity_detector_topic,
+            self.activity_callback,
+            1
+        )
+
+        self.object_detector_subscriber = self.create_subscription(
+            ObjectDetection3dSet,
+            self._object_detection_topic,
+            self.object_callback,
+            1
+        )
+
+        self.task_monitor_subscriber = self.create_subscription(
+            TaskUpdate,
+            self._task_monitor_topic,
+            self.task_callback,
+            1
+        )
+
+        # publisher
+        self.arui_update_publisher = self.create_publisher(
+            AruiUpdate,
+            self._arui_update_topic,
+            1
+        )
+
+        # message
+        self.arui_update_message = AruiUpdate()
+        self.arui_update_message.header.frame_id = "ARUI Update"
+
+        # lock
+        self.lock = Lock()
+
+        # detection uuids
+        self.uuids = dict()
+
+    def publish_update(self):
+        self.lock.acquire()
+
+        self.arui_update_message.header.stamp = self.get_clock().now().to_msg()
+        self.log.info(f"Publishing AruiUpdate: {self.arui_update_message}\n")
+        self.arui_update_publisher.publish(self.arui_update_message)
+
+        self.lock.release()
```

Comment:
We will want to use the context-manager usage of python locks. That version is advantageous in that if any context exit occurs (e.g. an exception), the lock will be released which prevents a deadlock on the next attempt to acquire a critical section.
```suggestion
        with self.lock:
            self.arui_update_message.header.stamp = self.get_clock().now().to_msg()
            self.log.info(f"Publishing AruiUpdate: {self.arui_update_message}\n")
            self.arui_update_publisher.publish(self.arui_update_message)
```